### ISSUE TITLE
bug(linter): no-const-assign false positive

### ISSUE THREAD

**Issue by Boshen** [oxc-project/oxc#2462]
```
  ! eslint(no-const-assign): Unexpected re-assignment of const variable getValue
    ,-[packages/rspack/src/util/memoize.ts:35:11]
 34 | export function memoizeValue<T>(fn: () => T): T {
 35 |     const getValue: () => any = memoize(fn);
    :           ^^^^|^^^
    :               `-- getValue is declared here as const
 36 |     return new Proxy({} as any, {
 37 |         get(_, property) {
 38 |             return getValue()[property];
 39 |         },
 40 |         set(_, property, newValue) {
 41 |             getValue()[property] = newValue;
    :             ^^^^|^^^
    :                 `-- getValue is re-assigned here
 42 |             return true;
    `----
```

https://github.com/web-infra-dev/rspack/actions/runs/7983718503/job/21799343618?pr=5727

**Comment by Dunqing** [comment]
This has been fixed in #2433

https://oxc-project.github.io/oxc/playground/?code=3YCAAIDeiYCAgICAgIC0G8rnONK89IyJ1j0iWUKhkRFk%2BNu1ORHxLupNGtxR36WYI2KLfY%2FeQ9EzlE%2F6i67jprobP31O24MdKbpWLda78GmqxCjPXp4ijQYisc%2FnKi1xBXDPLB%2BgmJXGWjbWDhaBuFP%2Bbq4nd%2BMES7zQoqdwuEGD1y%2B6lyVi6dCOfCznCvyFmLcZZCaRHpQXUxLiP7O0vVr1lPMlFeccyL%2F0hQX1FcZJPKdjE4ONLtmYOLUgcO6Yh0duxMKdDgPc5zdEjLIBL43FWKx0U9AGtxkn3LxomIr9am%2BsGWh4GGBG%2FFGaGneH1n53eKp2ItBF2Fy%2FVKcoM6CDL%2BLHYvRNyzRbEIkIVTuXbVr4Zb39px5J4xaTHROZxYV%2Ff7GS8hV0gUyJN1wuZCsv0BEm5Y4%2Fps5wrKUzOzjog90zwmTcdl6yKhHxQj%2FRRKd30%2FboR3ZdAwD3EHvcu%2BdgaROHhznHwvQQmu%2BEPBGyu83OrJYkcitPqfvbMnZ2axm3EWfj9CBxKFPfm6HaATZRaxl3L9617Tc%2FUAhTa3nTXTclmuxebXChZ0G3lqNrn4n67Fr8NDe6cMmw9L8blxnaMBZGPzFXk8ayn3ZpI5bgS4sHiMgy9vmH91LvMgbzZCzGdtDz4YWW5aJyQ1DxGAL7AX2YADBwld2uaba8Hmnkc%2F4SlYSoPzJtwposqwlqRJHDuMfG9M1ywunGylXbk9qLZRJyfG6zxsFze16qHHGik9U6kcg2iDXeKsrBQpU3A5DU%2FHrHcuihKKBv%2ByzInpRBAjGMwIi0e1hSf6rA4OQEf97vSw3rfsSXKPLgZBxBxRWT%2FBSuhBaf2w%2BrAq7xT9vo%2BvmCnXl0OA7HtHCoOAxdR34L3lZJazhiNkO3qvy4dvCnXfjpphEgHY1fjs%2BSaEWKjUolZffbNFmeQWT134JjEh2CEfJfHn0qr2suDFuz4S%2Frxe5%2B%2BUsmfx5oZufZ0GszRtCu6i0GvgqidHrxy%2FxmIAYdPVuvG3KoFyH2BDBEp%2B4Gf9lWy28YFBvE9XMMZWuhCg3iZxopZW0wt2IUWRr8FyAQtI%2B%2Fy7JlxWRUXKGMYKp%2BJzx8SwZNHRRIIvRbBIuXC3jKKrfQG%2FuTNxK%2BFfXTRxfoSFzNf2D1As4%3D

**Comment by Boshen** [comment]
Nice, I'll make a new release.
